docs/developers/analyzer-development.md

```diff
@@ -2,119 +2,342 @@
 hide:
   - footer
 ---
-# Write analyzers in Timesketch
+# Write analyzers for Timesketch
 
-## Multi Analyzer
+## Overview
 
-When you develop an analyzer that would benefit from creating smaller sub-jobs, you should use Multi Analyzer.
+This guide provides all steps to get you started with building a new Timesketch analyzer.
 
-For example The Sigma analyzer is such a Multi Analyzer. That means, the Sigma analyzer is calling `get_kwargs()` from [sigma_tagger.py](https://github.com/google/timesketch/blob/master/timesketch/lib/analyzers/sigma_tagger.py). That will return a list of all Sigma rules installed on the instance. The Main celery job then spawns one celery job per Sigma rule that can run in parallel or serial depending on the celery config and sizing of the Timesketch instance.
+## Background
 
-If `get_kwargs()` is not implemented in the analyzer, [tasks.py](https://github.com/google/timesketch/blob/master/timesketch/lib/tasks.py) expects it is not a multi analyzer, thus creating only one celery job.
+Timesketch analyzers are programs that run when new data is indexed, e.g. when
```

Comment:
Analyzers could also be run after the data is imported

Reply:
Good point. Will add the info here.